### ISSUE TITLE
Build tritonserver image failed due to hardcoded the”/tmp” folder  in build files 

### ISSUE THREAD

**Issue by yaoyunda** [triton-inference-server/server#3923]
**Description**
Build "tritonserver" from the code as [instruction](https://github.com/triton-inference-server/server/blob/main/docs/build.md#:~:text=Build%20without%20Docker.-,Building%20with%20Docker,-The%20easiest%20way).
It is going to fail. Here are the last console output:
error: make server failed
error: docker run tritonserver_builder failed.

From investigation it is found that the /tmp folder has not enough space which is under the root storage of Azure GPU VM.
Make a tmp folder under /data folder which is mounted on an extra disk with 1T GB space.
If the "tmp" is replaced with "data/tmp" in the file "build.py" and command line to build,
then the build is successful.

**Triton Information**
What version of Triton are you using?
r21.12

Are you using the Triton container or did you build it yourself?
build triton by myself

**To Reproduce**
Steps to reproduce the behavior.

1. Create a Azure GPU VM  with NvidiaGpuDriverLinux
2.  Prepare the working folder in VM and Install docker
3. In the working folder: git clone -b r21.12 https://github.com/triton-inference-server/server.git
4. cd server folder and run:
./build.py --cmake-dir=./build --build-dir=/tmp/citritonbuild --enable-logging --enable-stats --enable-tracing --enable-metrics --enable-gpu-metrics --enable-gpu --filesystem=gcs --filesystem=azure_storage --filesystem=s3 --endpoint=http --endpoint=grpc --repo-tag=common:r21.12 --repo-tag=core:r21.12 --repo-tag=backend:r21.12 --repo-tag=thirdparty:r21.12 --backend=ensemble --backend=tensorrt:r21.12 --backend=identity:r21.12 --backend=repeat:r21.12 --backend=square:r21.12 --backend=onnxruntime:r21.12 --backend=pytorch:r21.12 --backend=tensorflow1:r21.12 --backend=tensorflow2:r21.12 --backend=openvino:r21.12 --backend=python:r21.12 --backend=dali:r21.12 --backend=fil:r21.12 --repoagent=checksum:r21.12

Describe the models (framework, inputs, outputs), ideally include the model configuration file (if using an ensemble include the model configuration file for that as well).

**Expected behavior**
The "/tmp" folder should not be hardcoded in related files (build.py, Dockerfile.sdk, Dockerfile.QA, gen_qa_model_repository, gen_qa_custom_ops  ), expect a configurable folder name, and provide a way to configure.

**Comment by Tabrizian** [comment]
Thanks for bringing this to our attention. I have filed a ticket for this.

**Comment by CoderHam** [comment]
@yaoyunda the build.py allows the user to configure the build directory according to their needs: https://github.com/triton-inference-server/server/blob/main/build.py#L1467-L1473

**Comment by CoderHam** [comment]
@yaoyunda have you tried to repeat the build with a non-tmp directory for `--build-dir`?

**Comment by yaoyunda** [comment]
@CoderHam , yes, I did overwrite the build directory, this part is easy to handle, the issue is also spread to other files with hard coded "/tmp" folder, as I in the above "Expected behavior": (build.py, Dockerfile.sdk, Dockerfile.QA, gen_qa_model_repository, gen_qa_custom_ops ),5 files.

**Comment by CoderHam** [comment]
Thanks @yaoyunda. We will work on exposing a configurable directory so that you can choose to not use /tmp

**Comment by CoderHam** [comment]
@yaoyunda can you try using the --tmp-dir arg and building with https://github.com/triton-inference-server/server/pull/4029

I have removed the hardcoded /tmp directory usage.

**Comment by CoderHam** [comment]
@yaoyunda reminder to use the linked PR to confirm it unblocks you from your build.

**Comment by CoderHam** [comment]
Closed with https://github.com/triton-inference-server/server/pull/4029
